File: src/commands/add.ts
import { Command } from 'commander';
import chalk from 'chalk';
import inquirer from 'inquirer';
import { ConfigManager } from '../lib/config.js';
import { ProjectManager } from '../lib/project.js';
import { TickTickClient } from '../lib/api-client.js';

export const addCommand = new Command('add')
  .description('Add a new task')
  .argument('[title]', 'Task title (interactive mode if not provided)')
  .option('--desc <text>', 'Task description/content (alias: --content)')
  .option('--content <text>', 'Task description/content (alias: --desc)')
  .option('--due <date>', 'Due date (YYYY-MM-DD)')
  .option('--priority <number>', 'Priority (0-5)', parseInt)
  .option('--tags <tags>', 'Task tags (comma-separated)')
  .option('--project <id>', 'Project ID (overrides .ticktick file)')
  .action(async (title, options) => {
    try {
      // Load config
      const config = await ConfigManager.load();
      if (!ConfigManager.isAuthenticated(config)) {
        throw new Error('Not authenticated. Run \'ticktick auth login\' first');
      }

      // Determine project
      let projectId = options.project;
      let projectName = '';

      if (!projectId) {
        const ctx = await ProjectManager.getCurrentProjectContext();
        if (ctx) {
          projectId = ctx.file.projectId;
          projectName = ctx.file.projectName;
          console.log(chalk.cyan(`Using project from .ticktick: ${projectName}`));
        } else if (config.preferences.defaultProject) {
          projectId = config.preferences.defaultProject;
          console.log('Using default project from config');
        } else {
          throw new Error('No project specified. Run \'ticktick init\' or use --project flag');
        }
      }

      // Interactive mode if no title provided
      let taskTitle = title;
      let content = options.desc || options.content;
      let priority = options.priority;
      let dueDate = options.due;
      let tags: string[] | undefined = options.tags ? options.tags.split(',').map((t: string) => t.trim()) : undefined;

      if (!taskTitle) {
        console.log(chalk.cyan('\n📝 Create a new task (interactive mode)\n'));

        const answers = await inquirer.prompt([
          {
            type: 'input',
            name: 'title',
            message: 'Task title:',
            validate: (input: string) => input.trim().length > 0 || 'Title is required',
          },
          {
            type: 'input',
            name: 'content',
            message: 'Description (optional):',
          },
          {
            type: 'list',
            name: 'priority',
            message: 'Priority:',
            choices: [
              { name: 'None', value: 0 },
              { name: 'Low', value: 1 },
              { name: 'Medium', value: 3 },
              { name: 'High', value: 5 },
            ],
            default: 0,
          },
          {
            type: 'input',
            name: 'dueDate',
            message: 'Due date (YYYY-MM-DD, optional):',
            validate: (input: string) => {
              if (!input) return true;
              const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
              return dateRegex.test(input) || 'Date must be in YYYY-MM-DD format';
            },
          },
          {
            type: 'input',
            name: 'tags',
            message: 'Tags (comma-separated, optional):',
          },
        ]);

        taskTitle = answers.title;
        content = answers.content || undefined;
        priority = answers.priority;
        dueDate = answers.dueDate || undefined;
        tags = answers.tags ? answers.tags.split(',').map((t: string) => t.trim()).filter(Boolean) : undefined;
      }

      // Validate priority
      if (priority !== undefined && (priority < 0 || priority > 5)) {
        throw new Error('Priority must be between 0 and 5');
      }

      // Create API client
      const client = new TickTickClient(config.auth.accessToken);

      // Create task
      const task = await client.createTask({
        title: taskTitle,
        projectId,
        content,
        dueDate,
        priority,
        tags,
      });

      console.log(chalk.green.bold('\n✓ Task created successfully!'));
      console.log(`\nTitle: ${task.title}`);
      console.log(`ID: ${task.id}`);
      if (task.content) console.log(`Content: ${task.content}`);
      if (task.dueDate) console.log(`Due: ${task.dueDate}`);
      if (task.priority) console.log(`Priority: ${task.priority}`);
      if (task.tags && task.tags.length) console.log(`Tags: ${task.tags.join(', ')}`);
    } catch (error) {
      console.error(chalk.red('Error:'), (error as Error).message);
      process.exit(1);
    }
  });

File: internal/commands/add.go
package commands

import (
	"context"
	"fmt"
	"strings"

	"github.com/brettman/ticktickcli/internal/api"
	"github.com/brettman/ticktickcli/internal/config"
	"github.com/brettman/ticktickcli/internal/project"
	"github.com/fatih/color"
	"github.com/spf13/cobra"
)

// NewAddCmd creates the add command
func NewAddCmd() *cobra.Command {
	var (
		content     string
		dueDate     string
		priority    int
		tags        []string
		projectID   string
	)

	cmd := &cobra.Command{
		Use:   "add <title>",
		Short: "Add a new task",
		Long: `Add a new task to TickTick.

If run in a directory with a .ticktick file, the task will be added to that project.
Otherwise, you must specify a project with --project flag or it will use your default project.

Examples:
  ticktick add "Review pull request"
  ticktick add "Fix bug" --content "The login form is broken" --priority 3
  ticktick add "Deploy to production" --due 2026-01-15 --tags deploy,urgent
  ticktick add "Meeting notes" --project abc123`,
		Args: cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			title := args[0]
			return runAdd(title, content, dueDate, priority, tags, projectID)
		},
	}

	cmd.Flags().StringVar(&content, "content", "", "Task description/content")
	cmd.Flags().StringVar(&dueDate, "due", "", "Due date (YYYY-MM-DD)")
	cmd.Flags().IntVar(&priority, "priority", 0, "Priority (0-5, where 5 is highest)")
	cmd.Flags().StringSliceVar(&tags, "tags", []string{}, "Task tags (comma-separated)")
	cmd.Flags().StringVar(&projectID, "project", "", "Project ID (overrides .ticktick file)")

	return cmd
}

// runAdd handles adding a new task
func runAdd(title, content, dueDate string, priority int, tags []string, projectID string) error {
	// Validate priority
	if priority < 0 || priority > 5 {
		return fmt.Errorf("priority must be between 0 and 5")
	}

	// Load config and check authentication
	cfg, err := config.Load()
	if err != nil {
		return fmt.Errorf("failed to load config: %w", err)
	}

	if !cfg.IsAuthenticated() {
		return fmt.Errorf("not authenticated. Run 'ticktick auth login' first")
	}

	// Determine which project to use
	if projectID == "" {
		// Try to find .ticktick file
		ctx, err := project.GetCurrentProjectContext()
		if err == nil {
			projectID = ctx.File.ProjectID
			cyan := color.New(color.FgCyan)
			cyan.Printf("Using project from .ticktick: %s\n", ctx.File.ProjectName)
		} else if err == project.ErrNoTickTickFile {
			// Use default project from config
			if cfg.Preferences.DefaultProject != "" {
				projectID = cfg.Preferences.DefaultProject
				fmt.Printf("Using default project from config\n")
			} else {
				return fmt.Errorf("no project specified. Run 'ticktick init' or use --project flag")
			}
		} else {
			return fmt.Errorf("failed to load project context: %w", err)
		}
	}

	// Create API client
	client := api.NewClientFromConfig(
		cfg.Auth.ClientID,
		cfg.Auth.ClientSecret,
		cfg.Auth.AccessToken,
		cfg.Auth.RefreshToken,
		cfg.Auth.Expiry,
	)

	ctx := context.Background()

	// Create task request
	req := &api.CreateTaskRequest{
		Title:     title,
		ProjectID: projectID,
		Content:   content,
		DueDate:   dueDate,
		Priority:  priority,
		Tags:      tags,
	}

	// Create the task
	task, err := client.CreateTask(ctx, req)
	if err != nil {
		return fmt.Errorf("failed to create task: %w", err)
	}

	// Display success
	green := color.New(color.FgGreen, color.Bold)
	green.Println("\n✓ Task created successfully!")

	fmt.Printf("\nTitle: %s\n", task.Title)
	fmt.Printf("ID: %s\n", task.ID)
	if task.Content != "" {
		fmt.Printf("Content: %s\n", task.Content)
	}
	if task.DueDate != "" {
		fmt.Printf("Due: %s\n", task.DueDate)
	}
	if task.Priority > 0 {
		fmt.Printf("Priority: %s\n", formatPriority(task.Priority))
	}
	if len(task.Tags) > 0 {
		fmt.Printf("Tags: %s\n", strings.Join(task.Tags, ", "))
	}

	return nil
}

// formatPriority formats priority with visual indicator
func formatPriority(priority int) string {
	switch priority {
	case 5:
		return color.RedString("5 (Highest)")
	case 4:
		return color.RedString("4 (High)")
	case 3:
		return color.YellowString("3 (Medium)")
	case 2:
		return color.CyanString("2 (Low)")
	case 1:
		return color.CyanString("1 (Lowest)")
	default:
		return "0 (None)"
	}
}
